Microsoft.Toolkit.Uwp/Helpers/ObjectStorage/ApplicationDataStorageHelper.cs

```diff
@@ -0,0 +1,236 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Helpers;
+using Windows.Storage;
+using Windows.System;
+
+namespace Microsoft.Toolkit.Uwp.Helpers
+{
+    /// <summary>
+    /// Storage helper for files and folders living in Windows.Storage.ApplicationData storage endpoints.
+    /// </summary>
+    public class ApplicationDataStorageHelper : IFileStorageHelper, ISettingsStorageHelper
+    {
+        /// <summary>
+        /// Get a new instance using ApplicationData.Current and the provided serializer.
+        /// </summary>
+        /// <param name="objectSerializer">Serializer for converting stored values.</param>
+        /// <returns>A new instance of ApplicationDataStorageHelper.</returns>
+        public static ApplicationDataStorageHelper GetCurrent(Toolkit.Helpers.IObjectSerializer objectSerializer)
+        {
+            var appData = ApplicationData.Current;
+            return new ApplicationDataStorageHelper(appData, objectSerializer);
+        }
+
+        /// <summary>
+        /// Get a new instance using the ApplicationData for the provided user and serializer.
+        /// </summary>
+        /// <param name="user">App data user owner.</param>
+        /// <param name="objectSerializer">Serializer for converting stored values.</param>
+        /// <returns>A new instance of ApplicationDataStorageHelper.</returns>
+        public static async Task<ApplicationDataStorageHelper> GetForUserAsync(User user, Toolkit.Helpers.IObjectSerializer objectSerializer)
+        {
+            var appData = await ApplicationData.GetForUserAsync(user);
+            return new ApplicationDataStorageHelper(appData, objectSerializer);
+        }
+
+        /// <summary>
+        /// Gets the settings container.
+        /// </summary>
+        protected ApplicationData AppData { get; private set; }
+
+        private ApplicationDataContainer DefaultSettings => AppData.LocalSettings;
+
+        private StorageFolder DefaultFolder => AppData.LocalFolder;
+
+        private readonly Toolkit.Helpers.IObjectSerializer _serializer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApplicationDataStorageHelper"/> class.
+        /// </summary>
+        /// <param name="appData">The data store to interact with.</param>
+        /// <param name="objectSerializer">Serializer for converting stored values.</param>
+        public ApplicationDataStorageHelper(ApplicationData appData, Toolkit.Helpers.IObjectSerializer objectSerializer)
+        {
+            AppData = appData ?? throw new ArgumentNullException(nameof(appData));
```

Comment:
I expected most people to get an instance using the static methods instead of constructing it themselves, but I left it open to support scenarios where someone already has an ApplicationData object from somewhere else in their code and wants to plug it in here.

```
var storageHelper = ApplicationDataStorageHelper.GetCurrent(serializer);
var storageHelperForUser = await ApplicationDataStorageHelper.GetForUserAsync(user, serializer);
```

This feels familiar to how the ApplicationData API works already, so I kind of like the analog.

Reply:
I also wrote it with some wiggle room for someone to implement methods to access the other storage locations maintained under ApplicationData, such as `LocalCacheFolder`, `SharedLocalFolder`, and `TemporaryFolder`. That's why I made them helper properties.

Reply:
No worries, just checking. I had missed the `GetForUserAsync` method, so that makes more sense now. New feature! 😉

I guess that's a good thing for us to point out in docs about the difference of those locations and how that works. Like it's based on if the app is run on the user account vs. different users logging into the same app running under the same user account. Using the `Current` storage means everything saved would be available to both those users compared to if you passed in the user and segmented their data based on their account, right?

Reply:
You got it, `GetCurrent` returns a data store for the whole *application* (not user specific) and `GetForUserAsync` is for a single *user*.